packages/api/src/api/videos.ts

```diff
@@ -4,9 +4,17 @@ import logger from '../logger';
 
 export const videos = Router();
 
-videos.get('', async (req, res) => {
+videos.get('', async (req, res) => { 
+  const query = req.query.q || '';
+  if(typeof query !== 'string'){
+    res.status(400).send('query must be string');
+    return;
+  }
   try {
-    const videosList = await req.entityManager.find(Video, {});
+    const videosList = await req.entityManager.find(Video, { title: {
+      $ilike: `%${query.trim()}%`,
+    }
```

Comment:
We should only be passing the query in when it exists.
```suggestion
    // You'll need to figure out the appropriate type for the where object and then figure out whether `null` or `undefined` is acceptable as a second argument to the `find` clause;
    // To find the type, hover over the `find` method and you should see the type of the second argument
    const where: ??? | undefined = query ? {
      title: { 
        $ilike: `%${query.trim()}%`,
      }  
    } : undefined;
    const videosList = await req.entityManager.find(Video, where);
```